pallets/fiat_filter/src/tests.rs

```diff
@@ -0,0 +1,762 @@
+    });
+    ret
+}
+
+/// generate a random keypair
+pub fn gen_kp() -> sr25519::Pair {
+    sr25519::Pair::generate_with_phrase(None).0
+}
+/// get the latest block number
+pub fn block_no() -> u64 {
+    system::Module::<Test>::block_number()
+}
+// Create did for `did`. Return the randomly generated signing key.
+// The did public key is controlled by some non-existent account (normally a security
+// concern), but that doesn't matter for our purposes.
+pub fn create_did(origin: u64, did: did::Did) -> sr25519::Pair {
+    let kp = gen_kp();
+    let pubkey_bytes = did::Bytes32 {
+        value: kp.public().0,
+    };
+    let didpubkey = did::PublicKey::Sr25519(pubkey_bytes);
+    let key_detail = did::KeyDetail::new(did, didpubkey);
+    did::Module::<Test>::new(Origin::signed(origin), did, key_detail).unwrap();
+    kp
+}
+/// create a did with a random id and random signing key
+pub fn newdid(origin: u64) -> (did::Did, sr25519::Pair) {
+    let d: did::Did = rand::random();
+    (d, create_did(origin, d))
+}
+
+pub fn sign(payload: &StateChange, keypair: &sr25519::Pair) -> did::DidSignature {
+    did::DidSignature::Sr25519(did::Bytes64 {
+        value: keypair.sign(&payload.encode()).0,
+    })
+}
+
+/// create a random byte array with set len
+pub fn random_bytes(len: usize) -> Vec<u8> {
+    let ret: Vec<u8> = (0..len).map(|_| rand::random()).collect();
+    assert_eq!(ret.len(), len);
+    ret
+}
+
+fn measure_fees(call: TestCall) -> (u64, DispatchResultWithPostInfo) {
+    let balance_pre = <Test as Config>::Currency::free_balance(ALICE);
+    let executed = FiatFilterModule::execute_call(Origin::signed(ALICE), Box::new(call.clone()));
+    let balance_post = <Test as Config>::Currency::free_balance(ALICE);
+    let fees_paid_dock_permill = balance_pre - balance_post;
+    return (fees_paid_dock_permill, executed);
+}
+// fn exec(call: TestCall) -> DispatchResultWithPostInfo {
+//     return FiatFilterModule::execute_call(Origin::signed(ALICE), Box::new(call.clone()));
+// }
+
+// TESTS
+
+use core_mods::StateChange;
+use frame_support::{assert_noop, assert_ok};
+use frame_system::RawOrigin;
+
+mod tests_did_calls {
+    use super::*;
+    use did::{
+        Bytes32, Bytes64, DidRemoval, DidSignature, KeyDetail, KeyUpdate, PublicKey, DID_BYTE_SIZE,
+    };
+
+    #[test]
+    fn call_did_new() {
+        ext().execute_with(|| {
+            let d: did::Did = rand::random();
+            let kp = gen_kp();
+            let key_detail = did::KeyDetail::new(
+                d,
+                did::PublicKey::Sr25519(did::Bytes32 {
+                    value: kp.public().0,
+                }),
+            );
+
+            let call = TestCall::DidMod(did::Call::<Test>::new(d.clone(), key_detail));
+            let (fee_microdock, executed) = measure_fees(call);
+            assert_ok!(executed);
+
+            let pdi = executed.unwrap();
+            assert!(pdi.pays_fee == Pays::No);
+            const _50_CENTS: u64 = 22211363;
```

Comment:
Hardcoding `22211363` will make you update this if the fee in fiat changes. Better to use constants so updates happen in one place.